uamqp/session.py

```diff
@@ -239,7 +243,8 @@ def _outgoing_disposition(self, frame):
 
     def _incoming_disposition(self, frame):
         for link in self._input_handles.values():
-            link._incoming_disposition(frame)
+            if hasattr(link, '_incoming_disposition'):
```

Comment:
Rather than do this check - we should just add an empty `_incoming_disposition` function to the base link class that just does nothing. I think that might be more efficient than the hasattr check